src/equations/bbm_bbm_1d.jl

```diff
@@ -127,39 +127,43 @@ function source_terms_manufactured_reflecting(q, x, t, equations::BBMBBMEquation
     return SVector(dq1, dq2)
 end
 
-function create_cache(mesh,
-                      equations::BBMBBMEquations1D,
-                      solver,
-                      initial_condition,
-                      RealT,
-                      uEltype)
-    tmp1 = Array{RealT}(undef, nnodes(mesh)) # tmp1 is needed for the `RelaxationCallback`
+function create_cache(mesh, equations::BBMBBMEquations1D,
+                      solver, initial_condition,
+                      ::BoundaryConditionPeriodic,
+                      RealT, uEltype)
     D = equations.D
-    if solver.D1 isa PeriodicDerivativeOperator ||
-       solver.D1 isa UniformPeriodicCoupledOperator ||
-       solver.D1 isa PeriodicUpwindOperators
-        invImD2 = inv(I - 1 / 6 * D^2 * Matrix(solver.D2))
-        return (invImD2 = invImD2, tmp1 = tmp1)
-    elseif solver.D1 isa DerivativeOperator ||
-           solver.D1 isa UniformCoupledOperator ||
-           solver.D1 isa UpwindOperators
+    invImD2 = inv(I - 1 / 6 * D^2 * Matrix(solver.D2))
```

Comment:
In the long run, one should consider allowing more flexibility here. For example, FD methods should use sparse matrices and just a factorization instead of the inverse of a dense matrix. But that's not critical right now.

Reply:
Yeah, you are definitely right. IIRC, when I tried that some time ago, I had some problems making that properly work, but I don't remember the exact issue. Back then I focused on something that just works and is easy to implement andunderstand . But of course this is not the most performant way. I created #107 for that.